apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -99,10 +99,16 @@ def __eq__(self, other: object) -> bool:
 
     @property
     def uri(self) -> str:
+        """
+        Accessor for the object's storage URI
+        """
         return self._uri
 
     @abstractproperty
     def soma_type(self) -> str:
+        """
+        Abstract accessor for `soma_type` string. To be implemented by child classes.
```

Comment:
I don't think docstrings should have implementation details (e.g., 'to be implemented by ...').  They should focus on end-user (API user) information.

So, perhaps something like:  
> Return the SOMA object type, e.g., "SOMADataFrame"